artemis-core-client/src/main/java/org/apache/activemq/artemis/api/core/JsonUtil.java

```diff
@@ -212,10 +224,10 @@ public static void addToArray(final Object param, final JsonArrayBuilder jsonArr
       } else if (param instanceof Object[]) {
          JsonArrayBuilder objectArrayBuilder = JsonLoader.createArrayBuilder();
          for (Object parameter : (Object[])param) {
-            addToArray(parameter, objectArrayBuilder);
+            addToArray(parameter, objectArrayBuilder, ignoreInvalidParams);
          }
          jsonArrayBuilder.add(objectArrayBuilder);
-      } else {
+      } else if (!ignoreInvalidParams) {
```

Comment:
Instead of throwing an error,

what about just make the method be:

```java
   public static void addToObject(final String key, final Object param, final JsonObjectBuilder jsonObjectBuilder) {
      if (param instanceof Integer) {
         jsonObjectBuilder.add(key, (Integer) param);
      } else if (param instanceof Long) {
         jsonObjectBuilder.add(key, (Long) param);
      } else if (param instanceof Double) {
         jsonObjectBuilder.add(key, (Double) param);
      } else if (param instanceof String) {
         jsonObjectBuilder.add(key, (String) param);
      } else if (param instanceof Boolean) {
         jsonObjectBuilder.add(key, (Boolean) param);
      } else if (param instanceof Map) {
         JsonObject mapObject = toJsonObject((Map<String, Object>) param);
         jsonObjectBuilder.add(key, mapObject);
      } else if (param instanceof Short) {
         jsonObjectBuilder.add(key, (Short) param);
      } else if (param instanceof Byte) {
         jsonObjectBuilder.add(key, ((Byte) param).shortValue());
      } else if (param instanceof SimpleString) {
         jsonObjectBuilder.add(key, param.toString());
      } else if (param == null) {
         jsonObjectBuilder.addNull(key);
      } else if (param instanceof byte[]) {
         JsonArrayBuilder byteArrayObject = toJsonArrayBuilder((byte[]) param);
         jsonObjectBuilder.add(key, byteArrayObject);
      } else if (param instanceof Object[]) {
         final JsonArrayBuilder objectArrayBuilder = JsonLoader.createArrayBuilder();
         for (Object parameter : (Object[])param) {
            addToArray(parameter, objectArrayBuilder);
         }
         jsonObjectBuilder.add(key, objectArrayBuilder);
      } else if (param instanceof Number) {
         jsonObjectBuilder.add(key, ((Number)param).doubleValue());
      } else {
         jsonObjectBuilder.add(key, param.toString());
      }
   }
```


remove the argument ignoreInvalidParams


and do a similar change on the addToArray